frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.jsx

```diff
@@ -104,11 +124,67 @@ export function DashCardCardParameterMapper({
     _.isEqual(normalize(option.target), normalize(target)),
   );
 
+  const availableCardsWithoutEditingParameter = useMemo(() => {
```

Comment:
@oisincoveney all of this logic should be moved to redux actions. Maybe there won't be any component changes when you do so